addon/content/components/compose/composeWidget.jsx

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+import { composeActions } from "../../reducer/reducer-compose.js";
+import React from "react";
+import * as ReactRedux from "react-redux";
+import PropTypes from "prop-types";
+import { TextArea, TextBox } from "./composeFields.jsx";
+
+const INPUT_FIELDS = [
+  {
+    props: {
+      name: "from",
+      title: "message.fromHeader",
+      disabled: true,
+    },
+    component: TextBox,
+  },
+  {
+    props: {
+      name: "to",
+      title: "message.toHeader",
+      disabled: false,
+    },
+    component: TextBox,
+  },
+  {
+    props: {
+      name: "subject",
+      title: "compose.fieldSubject",
+      disabled: false,
+    },
+    component: TextBox,
+  },
+  {
+    props: {
+      name: "body",
+      disabled: false,
+    },
+    component: TextArea,
+  },
```

Comment:
Are there going to be a lot more of these? I wonder if it's worth the abstraction. Will `TextArea` and `TextBox` maintain the same props forever? If not, it seems weird to batch-create them when there are only four things.

Reply:
I'm not totally sure yet. The "from" text box will probably change into a drop-down of identities, the other ones we'll probably need to add some sort of autocomplete, and a few more boxes.

So they will get more complicated, even if they aren't now.

Reply:
I was confused when I originally wrote the response - yes I'll drop the abstraction, I don't think it is helping that much.